webapp/src/main/java/io/github/microcks/util/ai/AICopilotHelper.java

```diff
@@ -388,4 +388,100 @@ private static JsonNode followRefIfAny(JsonNode spec, JsonNode referencableNode)
    private static JsonNode getNodeForRef(JsonNode spec, String reference) {
       return spec.at(reference.substring(1));
    }
+
+  /**
+   * Some AI will reuse already present examples and x-microcks-operation in the
+   * spec. This method cleans a specification from its examples.
+   */
+  protected static String removeTagsFromOpenAPISpec(String specification, String operationName) throws Exception {
+    JsonNode specNode;
+    boolean isJson = specification.trim().startsWith("{");
+
+    if (isJson) {
+      specNode = JSON_MAPPER.readTree(specification);
+    } else {
+      specNode = YAML_MAPPER.readTree(specification);
+    }
+
+    // Resolve schemas and Remove examples recursively from the root.
+    resolveReferenceAndRemoveTagsInNode(specNode, specNode);
+    reduceSpecSize(specNode, operationName);
```

Comment:
Wouldn't be more efficient to switch both method invocations? Something like:
```java
reduceSpecSize(specNode, operationName);
resolveReferenceAndRemoveTagsInNode(specNode, specNode);
```
It may avoid resolving references and removing spec elements in operations that would have been dropped later on.

Reply:
I understand, but it is necessary that the resolveReferenceAndRemoveTagsInNode method (now renamed to resolveReferenceAndRemoveTokensInNode) be executed first because in this method the schema references are resolved, this is necessary before removing the "components" token.